File: app/src/main/java/com/example/project1/ListAdapter.java
package com.example.project1;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class ListAdapter extends RecyclerView.Adapter<ListAdapter.MyViewHolder> {
    private ArrayList<Restaurant> mDataset;

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        // each data item is just a string in this case
        public TextView name;
        public TextView number;
        public ImageView cost;
        public MyViewHolder(View v) {
            super(v);
            name = v.findViewById(R.id.txt_name_rest);
            number = v.findViewById(R.id.txt_number_rest);
        }
    }

    // Provide a suitable constructor (depends on the kind of dataset)
    public ListAdapter(ArrayList<Restaurant> myDataset) {
        mDataset = myDataset;
    }

    @NonNull
    @Override
    public ListAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // create a new view
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recycle_row, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;

    }

    @Override
    public void onBindViewHolder(@NonNull ListAdapter.MyViewHolder holder, int position) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        Restaurant rest = mDataset.get(position);
        holder.name.setText(rest.getName());
        holder.number.setText(rest.getNumber()+"");
    }

    @Override
    public int getItemCount() {
        return mDataset.size();

    }
}

File: app/src/main/java/com/example/project1/fragments/ListAdapter.kt
package com.example.project1.fragments

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.project1.R
import com.example.project1.Utils.ImageDownloader
import com.example.project1.model.Restaurant
import com.squareup.picasso.Picasso
import java.util.ArrayList


class ListAdapter// Provide a suitable constructor (depends on the kind of dataset)
    (private val mDataset: ArrayList<Restaurant>, val callback: () -> Unit) : RecyclerView.Adapter<ListAdapter.MyViewHolder>() {

    class MyViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        // each data item is just a string in this case
        var name: TextView
        var number: TextView
        var image: ImageView?= null
        var cost: TextView?= null

        init {
            name = v.findViewById(R.id.txt_name_rest)
            number = v.findViewById(R.id.txt_number_rest)
            image = v.findViewById(R.id.img_logo_rest)
            cost = v.findViewById(R.id.img_cost_rest)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        // create a new view
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.recycle_row, parent, false)

        return MyViewHolder(v)

    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        // - get element from your dataset at this position
        // - replace the contents of the view with that element
        val rest = mDataset[position]
        holder.name.text = rest.name
        holder.number.text = rest.contactInfo.toString() + ""
        if(rest.images.isNotEmpty())
        {
            Picasso.get().load(rest.images[0]).error(R.drawable.logo_res).into(holder.image)
        }
        if(rest.cost==0)
        {
           holder.cost?.text = "$"
        }
        if(rest.cost==1)
        {
            holder.cost?.text = "$$"
        }
        holder.itemView.setOnClickListener {
            callback()
        }
    }

    override fun getItemCount(): Int {
        return mDataset.size

    }
}
